src/components/Modal/index.js

```diff
@@ -1,69 +1,98 @@
+    this.state = {isOpen: false}
+  }
 
-  const containerEl = document.getElementById(options.containerId)
+  componentDidMount() {
+    const { buttonId } = this.props
+    console.log("componentDidMount",buttonId)
+    const button = document.getElementById(buttonId)
+    if (!button){
+      console.warn(`The button with id #${buttonId} cannot be found`)
+      return
+    }
+    button.addEventListener('click', this.openModal)
+    button.disabled = false
+    this.setState({button})
+  }
 
-  if (options.useModal) {
-    options.mount = containerEl
-    options.mount.parentNode.insertBefore(modal, options.mount.nextSibling)
+  componentWillUnmount() {
+    const { button } = this.state
+    if (button) button.removeEventListener('click', this.openModal)
   }
-  else {
-    options.mountRoot = containerEl
-    options.mount = modal.getElementsByClassName(style.content)[0]
-    options.mountRoot.appendChild(modal)
+
+  openModal = () => {
+    events.emit('onBeforeOpen')
+    this.setState({isOpen: true})
   }
 
-  events.emit('modalMounted', options)
-}
+  onAfterOpen = () => events.emit('onOpen')
 
-const vanillaOptions = {
-  loadClass: style.vanillaModal,
-  class: style.modalVisible,
-  modal: '.'+style.modal,
-  modalInner: '.'+style.inner,
-  modalContent: '.'+style.content,
-  onOpen: () => events.emit('onOpen'),
-  onClose: () => events.emit('onClose'),
-  onBeforeOpen: () => events.emit('onBeforeOpen'),
-  onBeforeClose: () => events.emit('onBeforeClose')
-}
-console.log(vanillaOptions)
-
-events.on('modalMounted', (options) => {
-  if (options.useModal) {
-    const modal = new VanillaModal(vanillaOptions)
-    const button = document.getElementById(options.buttonId)
-    const id = "#"+options.containerId
-    button.addEventListener('click', () => modal.open(id))
-    events.on('onOpen', () => options.mount.style.display = 'block')
-    events.on('onClose', () => options.mount.style.display = 'none')
-    events.on('closeModal', () => modal.close())
-    events.once('ready', () => button.disabled = false)
+  onRequestClose = () => {
+    console.log("onRequestClose")
+    events.emit('onBeforeClose')
+    this.setState({isOpen: false})
   }
-  else {
-    events.emit('onBeforeOpen')
-    events.emit('onOpen')
-    options.mountRoot.style.display = 'block'
+
+  onAfterClose = () => events.emit('onClose')
+
+  //TODO get rid of preact or wait for a new version that supports pure components properly
+  //ReactModal is a standard React Component, therefore we have to force preact-compat on it
+  //the problem is that preact still doesn't support pure components, so we have to impurify its child
+  static WrapperContentImpure = impurify(WrapperContent)
+
+  render () {
+    return (
+      <ReactModal
+        isOpen={this.state.isOpen || this.props.isOpen}
+        onAfterOpen={this.onAfterOpen}
+        onRequestClose={this.onRequestClose}
+        onAfterClose={this.onAfterClose}
+        portalClassName={style.portal}
```

Comment:
all the classes used by `ReactModal` are namespaced. This required changes in that npm module, hence why it was forked (here: https://github.com/rfreitas/react-modal) and published as `react-modal-onfido`